tests/integration/Filters.test.js

```diff
@@ -0,0 +1,331 @@
+    });
+
+    it("closes dialog window on close window", async () => {
+
+        wrapper = await mountMenu(dishes);
+
+        let filterButton = wrapper.find("button.mx-2");
+
+        expect(filterButton.exists()).toBeTruthy();
+        expect(filterButton.find("i.mdi-filter")).toBeTruthy();
+
+        // Open Dialog
+        await filterButton.trigger("click");
+
+        // Click Filter Dishes Btn
+        let closeFiltersBtn = wrapper.find("#dialog-close-btn");
+        await closeFiltersBtn.trigger("click");
+
+        let dialog = wrapper.find(".v-dialog");
+
+        expect(dialog.classes('v-dialog--active')).toBe(false);
+    });
+
+    it("it clears filters and shows all default items", async () => {
+
+        wrapper = await mountMenu(dishes);
+
+        // Open dialog box first
+        let filterButton = wrapper.find("button.mx-2");
+        await filterButton.trigger("click");
+        
+        // range slider is not compatible with tests; it has to be bypassed
+        let filters = wrapper.findComponent(Filters);
+
+        // Default values are 15, 200
+        filters.vm.$data.range = [15,70];
+
+        // Filter Dishes
+        let filterDishesBtn = wrapper.find("#filter-dishes-btn");
+        await filterDishesBtn.trigger("click");
+
+        // Clear Filters
+        let clearFiltersBtn = wrapper.find("#clear-filters-btn");
+        await clearFiltersBtn.trigger("click");
+        
+        
+        let allDishes = wrapper.findAll(".all-items").length;
+        let filteredDishes = wrapper.findAll(".filtered-items .dish-inner").length;
+
+        expect(allDishes).toBe(1);
+        expect(filteredDishes).toBe(0);
+    });
+
+    it("it changes filter icon when it has filters", async () => {
+
+        wrapper = await mountMenu(dishes);
+
+        // Open dialog box first
+        let filterButton = wrapper.find("#filter-button");
+        await filterButton.trigger("click");
+        
+        // range slider is not compatible with tests; it has to be bypassed
+        let filters = wrapper.findComponent(Filters);
+
+        // Default values are 15, 200
+        filters.vm.$data.range = [15,70];
+
+        // Filter Dishes
+        let filterDishesBtn = wrapper.find("#filter-dishes-btn");
+        await filterDishesBtn.trigger("click");
+
+        let clearFiltersBtn = wrapper.find("#clear-filters-btn");
+
+        expect(filterButton.isVisible()).toBe(false);
+        expect(clearFiltersBtn.isVisible()).toBe(true);
+    
+    });
+    it("it sorts by price desc", async () => {
+        wrapper = await mountMenu(dishes);
+
+        // Open dialog box first
+        let filterButton = wrapper.find("#filter-button");
+        await filterButton.trigger("click");
+        
+        // range slider is not compatible with tests; it has to be bypassed
+        let filters = wrapper.findComponent(Filters);
+
+        // Default values are 15, 200
+        filters.vm.$data.range = [15,200];
```

Comment:
Don't apply the filters. It it stops working, you'll get a false negative in this test

Reply:
Resolved